localstack/services/secretsmanager/secretsmanager_starter.py

```diff
@@ -1,122 +1,13 @@
-
-            return json.dumps(result)
-        else:
-            raise SecretNotFoundException()
-
-    setattr(SecretsManagerBackend, "get_resource_policy", get_resource_policy_model)
-
-    def get_resource_policy_response(self):
-        secret_id = self._get_param("SecretId")
-        return secretsmanager_backends[self.region].get_resource_policy(secret_id=secret_id)
-
-    setattr(SecretsManagerResponse, "get_resource_policy", get_resource_policy_response)
-
-    def delete_resource_policy_model(self, secret_id):
-        if self._is_valid_identifier(secret_id):
-            self.secrets[secret_id].policy = None
-            return json.dumps(
-                {
-                    "ARN": self.secrets[secret_id].arn,
-                    "Name": self.secrets[secret_id].secret_id,
-                }
-            )
-        else:
-            raise SecretNotFoundException()
-
-    if not hasattr(SecretsManagerBackend, "delete_resource_policy"):
-        setattr(
-            SecretsManagerBackend,
-            "delete_resource_policy",
-            delete_resource_policy_model,
-        )
-
-    def delete_resource_policy_response(self):
-        secret_id = self._get_param("SecretId")
-        return secretsmanager_backends[self.region].delete_resource_policy(secret_id=secret_id)
-
-    if not hasattr(SecretsManagerResponse, "delete_resource_policy"):
-        setattr(
-            SecretsManagerResponse,
-            "delete_resource_policy",
-            delete_resource_policy_response,
-        )
-
-    def put_resource_policy_model(self, secret_id, resource_policy):
-        policy_validator = IAMPolicyDocumentValidator(resource_policy)
-        policy_validator._validate_top_elements()
-        policy_validator._validate_version_syntax()
-        if self._is_valid_identifier(secret_id):
-            self.secrets[secret_id].policy = resource_policy
-            return json.dumps(
-                {
-                    "ARN": self.secrets[secret_id].arn,
-                    "Name": self.secrets[secret_id].secret_id,
-                }
-            )
-        else:
-            raise SecretNotFoundException()
-
-    if not hasattr(SecretsManagerBackend, "put_resource_policy"):
-        setattr(SecretsManagerBackend, "put_resource_policy", put_resource_policy_model)
-
-    def put_resource_policy_response(self):
-        secret_id = self._get_param("SecretId")
-        resource_policy = self._get_param("ResourcePolicy")
-        return secretsmanager_backends[self.region].put_resource_policy(
-            secret_id=secret_id, resource_policy=json.loads(resource_policy)
-        )
-
-    if not hasattr(SecretsManagerResponse, "put_resource_policy"):
-        setattr(SecretsManagerResponse, "put_resource_policy", put_resource_policy_response)
-
-
 def start_secretsmanager(port=None, asynchronous=None, backend_port=None, update_listener=None):
     apply_patches()
     result = start_moto_server(
```

Comment:
The goal would be to remove this function (essentially this entire file), as `start_moto_server(..)` will no longer be required here after this migration.. 🎉

Reply:
with the patch utility, i think it's even ok to apply patches directly on module-level, due to lazy loading :-) like here: https://github.com/localstack/localstack/blob/bb2a1038716f10f4648234fc2ba55ae479395601/localstack/services/acm/provider.py

Reply:
I have tried that, but am unable to make it work with patches adding attributes to the classes?

Reply:
yes good point, that's true. "setattr if not hasattr` pattern of patching is actually not supported by the patch utility. could be interesting to add but may not be straight forward.